core/src/main/java/com/linecorp/armeria/internal/common/DefaultSplitHttpResponse.java

```diff
@@ -55,9 +62,12 @@ public final CompletableFuture<ResponseHeaders> headers() {
     private static final class SplitHttpResponseBodySubscriber extends SplitHttpMessageSubscriber {
 
         private final HeadersFuture<ResponseHeaders> headersFuture = new HeadersFuture<>();
+        private final Predicate<ResponseHeaders> responseHeadersPredicate;
```

Comment:
What do you think about adding Javadoc for this field? It's a little bit unclear for me to understand what this exactly does from the first glance.